app/javascript/react/components/GiraffeShowPage.js

```diff
@@ -29,17 +32,57 @@ const GiraffeShowPage = (props) => {
       .catch((error) => console.error(`Error in fetch: ${error.message}`));
   }, []);
 
+  const addNewReview = (formPayLoad) => {
+    fetch(`/api/v1/giraffes/${id}/reviews`, {
+      credentials: "same-origin",
+      method: "POST",
+      body: JSON.stringify(formPayLoad),
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response;
+        } else {
+          let errorMessage = `${response.status} (${response.statusText})`,
+            error = new Error(errorMessage);
+          throw error;
+        }
+      })
+      .then((response) => response.json())
+      .then((body) => {
+        if (body.review) {
+          setGiraffe({
+            ...giraffe,
+            reviews: [...giraffe.reviews, body.review],
+          });
+        } else if (body.errors) {
+          setErrors(body.errors);
+        } else {
+          console.error("ERROR: Unexpected server response");
+        }
```

Comment:
Is there a situation where we will hit the `else` on 63 and not the `.catch` on 67?  I think we may be able to just remove the else conditional 63-65

Reply:
removed redundant else block